DataReader/Controller/ViewController.swift

```diff
@@ -43,125 +40,42 @@ class ViewController: UIViewController {
     
     override func motionEnded(_ motion: UIEventSubtype, with event: UIEvent?) {
         if motion == .motionShake {
-            self.clearData()
+            coreDataStack.clearData()
             self.activitiIndicator.isHidden = false
-            APIService.requestData(delegate: self)
+            let jsonUrlString = "https://private-5e934f-datatextapi.apiary-mock.com/data"
+            guard let url = URL(string: jsonUrlString) else { return }
+
+            self.apiService.requestData(url: url) { (data, error) -> Void in
+                guard let data = data else { return }
+                DispatchQueue.main.async {
+                    self.coreDataStack.saveInCoreDataWith(data: data)
```

Comment:
using self here is dangerous... 
consider scenario:what if view controller won't be visible? for example user jumped to another VC

Reply:
done